### ISSUE TITLE
Symlinks in /boot?

### ISSUE THREAD

**Issue by nmeum** [voidlinux/void-packages#972]
It would be kind of helpful if there would be a symlinks for the most recent kernel and initramfs version in `/boot`. A symlink with a '.old' prefix could point to the previous kernel/initramfs version.

This would make it a lot easier to configure bootloaders which are not grub, because you could just point them to the symlink instead of changing their configuration with a kernel hook after each kernel upgrade. The following packages would profit from this:

- gummiboot
- LILO
- Syslinux

This symlink feature would be trivial to realize, you just need to create one post-install and one post-remove kernel hook. The question is: a) Is this considered a good idea by anyone else? b) If so: Which package should ship these two hooks?

**Comment by chneukirchen** [comment]
+1, this would also be useful for people that use Grub with a static configuration.

**Comment by nmeum** [comment]
Yeah right. I would be willing to create a pull request for this, I just need to know which package should ship that hooks…

**Comment by xtraeme** [comment]
The FAT filesystem does not support links, and /boot on EFI these days must be FAT... there's no other way than copying in that case.

**Comment by nmeum** [comment]
Well…there is a simple solution to this:

```shell
filesystem="$(mount | grep /boot | awk '{print $5}')"
if [ filesystem = "FAT" ]; then
  cp # ...
else
  ln -s # ...
fi
```

**Comment by Vaelatern** [comment]
This issue should be closed?

**Comment by pullmoll** [comment]
The symbolic link would be in the root file system (/) which is always a fs that supports symlinks.
The target being on a non-symlink-capable fs does not matter in this case.

**Comment by chneukirchen** [comment]
I think this feature would still be useful, even if it failed on fat32.

There are plenty of ways to use EFI and still have your kernel on a real file system.
